fv3_cap.F90

```diff
@@ -0,0 +1,1691 @@
+                                    exportFieldShare,                        &
+                                    nImportFields,    importFields,          &
+                                    importFieldsList, importFieldTypes,      &
+                                    importFieldShare, importFieldsValid,     &
+                                    queryFieldList,   fillExportFields,      &
+                                    exportData
+  use module_cap_cpl,         only: realizeConnectedCplFields,               &
+                                    clock_cplIntval, diagnose_cplFields
+
+
+  implicit none
+  private
+  public SetServices
+!
+!-----------------------------------------------------------------------
+!
+  type(ESMF_Clock),save                       :: clock_fv3
+
+  type(ESMF_GridComp)                         :: fcstComp
+  type(ESMF_State)                            :: fcstState
+  character(len=esmf_maxstr),allocatable      :: fcstItemNameList(:)
+  type(ESMF_StateItem_Flag), allocatable      :: fcstItemTypeList(:)
+  type(ESMF_FieldBundle),    allocatable      :: fcstFB(:)
+  integer, save                               :: FBCount
+
+  type(ESMF_GridComp),    allocatable         :: wrtComp(:)
+  type(ESMF_State),       allocatable         :: wrtState(:)
+  type(ESMF_FieldBundle), allocatable         :: wrtFB(:,:)
+
+  type(ESMF_RouteHandle), allocatable         :: routehandle(:,:)
+  integer, allocatable                        :: fcstPetList(:)
+
+  logical :: profile_memory = .true.
+
+  character(len=160) :: nuopcMsg
+  integer            :: timeslice = 0
+  integer            :: fcstmype
+  integer            :: dbug = 0
+
+!-----------------------------------------------------------------------
+
+  contains
+
+  !-----------------------------------------------------------------------
+  !------------------- Solo fv3gfs code starts here ----------------------
+  !-----------------------------------------------------------------------
+
+  subroutine SetServices(gcomp, rc)
+
+    type(ESMF_GridComp)  :: gcomp
+    integer, intent(out) :: rc
+    character(len=*),parameter  :: subname='(fv3gfs_cap:SetServices)'
+
+    rc = ESMF_SUCCESS
+
+    ! the NUOPC model component will register the generic methods
+    call NUOPC_CompDerive(gcomp, model_routine_SS, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+    ! initialization, switching to IPD versions
+    call ESMF_GridCompSetEntryPoint(gcomp, ESMF_METHOD_INITIALIZE, &
+                                    userRoutine=InitializeP0, phase=0, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+    ! set entry point for methods that require specific implementation
+    call NUOPC_CompSetEntryPoint(gcomp, ESMF_METHOD_INITIALIZE, &
+                                 phaseLabelList=(/"IPDv01p1"/), userRoutine=InitializeAdvertise, rc=rc)
```

Comment:
`IPDvXXpY` syntax is no longer required for registering methods in SetServices.  Simpler syntax available in ESMF 8.1: http://earthsystemmodeling.org/docs/release/ESMF_8_1_1/NUOPC_refdoc/node4.html#NUOPC_Model

Reply:
I am interested on how the ESMF is setting up threading for each subcomponent, is there any interface change for that?